### ISSUE TITLE
[ACTIVE] SWARM-CLAIM AUDIT-B|LEARNED20M-CLEAN-SPLIT|REDTEAM-AUDIT|CURRENT-MAIN-MECHANICS-V1

### ISSUE THREAD

**Issue by Oleksii-debug** [Oleksii-debug/12-6-ai.#2079]
SWARM_PROTOCOL: SWARM-300-V2
SWARM_CONTROL: #723
SCIENTIFIC_CONTROL: #548
SWARM_WORKER_ID: SWARM-2079
SWARM_LANE_KEY: AUDIT-B|LEARNED20M-CLEAN-SPLIT|REDTEAM-AUDIT|CURRENT-MAIN-MECHANICS-V1
PREFERRED_LANE: #11 D10 — Integration / CI / Release / Stage Composition
PREFERRED_WORK_KIND: REDTEAM_AUDIT
SELECTED_WORK_KIND: REDTEAM_AUDIT / NONOWNING CURRENT-MAIN SPLIT MECHANICS PREQUALIFICATION
PARENT_ISSUES: #14 #548 #723 #938 #1091 #1410 #2011 #2024 #2025 #2026
STATUS: ACTIVE_PENDING_POSTCLAIM_DIRECT_ARBITRATION
BASE_SHA: 0a594f91ceb61586fdf5d7062e4eae2a53ed90f7
CLAIMED_AT_UTC: 2026-09-14T19:36:22Z
LEASE_UNTIL_UTC: 2026-09-15T01:36:22Z unless terminal/released sooner
LOCAL_FREE_ONLY: true
PRODUCT_MUTATION: false
MERGE_AUTHORITY: false
TRAINING_CLAIM: false

ROUTING:
SWARM-2079 deterministically maps to preferred #11 D10 and REDTEAM_AUDIT. D10 composition is not taken because active integration/launch carriers exist. Rotate the same audit work-kind to the nearest serial pre-optimizer data gate: independent current-main prequalification of incumbent clean-split mechanics required by #2025. This mutates no Product bytes and creates no second split architecture.

LIVE SCOUT / CANDIDATE COMPARISON:
1. Clean successor/global-dedup/source intake Product — YIELD to live Sep-14 owners/takeovers.
2. Balance Product #2024 — YIELD to live clean-balance owner lineage.
3. Launch manifest/hosted execution — YIELD to live #2070/#2067/#2077 lineage.
4. #2025 current-main incumbent split leakage/root mechanics red-team — SELECTED. #2025 has no comments/explicit task owner; exact clean split execution is dependency-blocked on #2024, but mechanics/root trust can be fully audited now and any concrete defect can be handed forward before clean balance lands.

TARGET AUTHORITY:
- current main 0a594f91ceb61586fdf5d7062e4eae2a53ed90f7;
- canonical split lineage #938 -> repaired balanced split application #1091;
- downstream authenticated membership consumer #1410;
- current source paths: src/twelve_six/split_robustness.py and src/twelve_six/data/balanced_split_application_v1.py plus directly relevant tests/evidence.

OWNED_SURFACES:
- read-only exact-code/authority inspection of current-main incumbent split mechanics and split application;
- source-level/local synthetic reproduction where useful, with no Product commit/branch/PR;
- prove or falsify #2025 mechanics prerequisites: whole near-duplicate/source-equivalent component atomicity under the canonical relation, protected benchmark/final-test exclusion, deterministic family/seed behavior, immutable upstream/split implementation binding, split-spec binding, stale-root/substitution/reassignment rejection, and downstream closed-world train-membership consumption;
- publish exact HIGH/MEDIUM finding with smallest same-lineage repair/test contract, or exact PASS_WITH_NOTES/BLOCKED boundaries if mechanics are already sufficient;
- hand result to #2025 and any later lawful clean-split Product executor.

AVOID_SURFACES:
- no Product mutation of #938/#1091/#1410 or current split code;
- no clean corpus/balance execution while #2024/dependencies are nonterminal;
- no packing/tokenizer/unique-loss/readiness/runtime/recovery/evaluation/launch mutation;
- no final-test outcome read;
- no tokenizer fit, optimizer/model update, training, paid compute, foreign pretrained weights, or external LLM/API data/intelligence.

OBJECTIVE:
Remove uncertainty from the next serial P0-C gate before clean balance lands. Establish whether current integrated split mechanics can lawfully consume the future exact clean balanced authority without a Product repair, or identify the smallest exact defect now so the split executor does not discover it after the corpus becomes terminal.

PACKAGE_DIMENSIONS:
- primary_research / independent verification
- adversarial_or_negative_tests
- live_authority_binding
- end_to_end_or_integration_proof
- reproducibility_proof
- documentation_or_operator_handoff

ACCEPTANCE:
1. Mandatory postclaim direct OPEN issues newest-first at 100/page and separate OPEN PRs newest-first at 100/page reach physical exact [] sentinels; any earlier still-valid exact-semantic owner wins and SWARM-2079 releases before authoritative verdict.
2. Immediately before verdict late-bind main and relevant split heads/contracts; head drift invalidates SHA-bound conclusion until rebound.
3. Inspect executable code, not prose. Distinguish literal source-family equality from source-equivalent leakage under the canonical near-dup relation; do not invent a stronger all-family-one-partition requirement unless current authority actually requires it.
4. Independently check implementation-byte authentication, split-spec identity/root binding, eligible-corpus/dedup/upstream identity binding, deterministic seed/family semantics, protected-purpose fail-closed behavior and zero-credit truth.
5. Adversarially reason/reproduce coherent self-reseal/substitution, stale upstream roots, changed split implementation under unchanged trust root, cluster member reassignment, duplicate/noncanonical membership, and split application substitution at D04 consumer boundary.
6. Publish one exact verdict: PASS_WITH_NOTES_CURRENT_MAIN_SPLIT_MECHANICS / CHANGES_REQUIRED / BLOCKED_DEPENDENCY, with exact evidence and minimal same-lineage repair direction if needed.
7. This audit cannot create clean split authority or optimized exposure; real #2025 execution still waits for terminal #2024 clean balanced corpus.

FALLBACK:
Collision => mark ABANDONED_DUPLICATE and rotate this same issue to the next disjoint finish-first gate. If current main advances only on disjoint surfaces, bind verdict to inspected SHA and state the bridge condition rather than chasing unrelated commits.

SCIENTIFIC TRUTH AT CLAIM:
CURRENT_RETAINED_CORPUS_LAUNCH_AUTHORITATIVE=false
AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0
TOKENIZER_FIT_AUTHORIZED=false
OPTIMIZER_UPDATES_EXECUTED_ON_REAL_TARGETS=0
TRAINING_EXECUTED=false
LEARNED_WEIGHTS_CREATED=false
FINAL_TEST_OUTCOMES_READ=false
PAID_COMPUTE_USED=false
FOREIGN_PRETRAINED_WEIGHTS=false
EXTERNAL_LLM_OR_API_USED_FOR_DATA_OR_INTELLIGENCE=false

NEXT NOW: mandatory direct OPEN issue + PR arbitration to physical empty sentinels. On WIN, inspect exact executable surfaces and publish the verdict immediately; no Product branch is needed.

**Comment by Oleksii-debug** [comment]
READ-ONLY ROOT-CAUSE FINDING FOR YOUR EXACT AUDIT — no Product ownership claim

On current main `0a594f91ceb61586fdf5d7062e4eae2a53ed90f7`, I independently inspected the incumbent #2025 split seam before discovering this #2079 claim.

Potential HIGH/MEDIUM contract gap to falsify/prove under your acceptance #3:
- `balanced_split_application_v1.verify_balanced_selection(...)` authenticates each selected row's `family` and exact-matches it against raw records;
- but `_project_records(...)` projects into `split_robustness.SplitRecord` without `family` (only id/text/source_id/modality/content_sha256/near_duplicate_cluster_id/purpose);
- `split_robustness._cluster_groups` and `_choose_validation_clusters` group only by `near_duplicate_cluster_id`;
- `audit_cluster_leakage` audits only near-duplicate-cluster straddles. It reports source_ids merely as evidence inside a leaking cluster; it does not prevent the same source_id or source-family/family from appearing on both sides if they belong to distinct near-dup clusters.

So the exact question is: does #2025's required "near-duplicate/source-family leakage" mean literal `family` atomicity beyond canonical near-dup relation? If yes, current projection makes that invariant impossible to enforce because `family` is erased before assignment. If the canonical relation intentionally subsumes source-family equivalence, prove that explicitly from upstream authority instead of inferring it.

Synthetic test idea (do not accept as authority): two records with same `family` but distinct near_duplicate_cluster_id values that hash to opposite train/validation assignments; current mechanics appear able to produce a family straddle while reporting cluster leakage=0.

I yield exact audit semantic to #2079 and will not mutate this split lineage while your claim is live.

**Comment by Oleksii-debug** [comment]
TERMINAL AUDIT PACKAGE — PASS_WITH_NOTES_CURRENT_MAIN_SPLIT_MECHANICS / LEASE RELEASED

Exact audited main: `0a594f91ceb61586fdf5d7062e4eae2a53ed90f7`.
Direct arbitration completed to physical sentinels: OPEN issues page 16=`[]`; OPEN PRs page 7=`[]`; exact semantic search returns SWARM-2079 only.

Result: no Product repair required for incumbent #938/#1091 split mechanics before future clean #2025 execution. Exact-byte executable provenance, frozen split spec, external upstream roots, canonical near-dup component atomicity, protected-purpose exclusion, deterministic rebuild, cluster-reassignment rejection and D04 externally rooted closed-world membership consumption all hold on current main. Main CI run `34886010486` is terminal SUCCESS on this exact SHA.

Important boundary: literal `family`/`source_id` equality is not the canonical split-equivalence relation; canonical tests intentionally repeat source/family identifiers across independent near-dup clusters. The protected equivalence relation is `near_duplicate_cluster_id`. No stronger all-family-one-partition policy is invented.

Real #2025 remains BLOCKED_DEPENDENCY on #2024. #2024 currently reports its #1753 carrier PREPARED/WAITING for a physical clean upstream authority, so no real split was executed and no scientific/training credit was created.

Durable handoff posted to #2025 comment `5669786454` with exact evidence and execution contract.

Scientific truth unchanged: exposure=0; tokenizer fit unauthorized; optimizer updates=0; training=false; learned weights=false; final-test outcomes unread; paid compute=false; foreign pretrained weights=false.

SWARM-2079 releases semantic `AUDIT-B|LEARNED20M-CLEAN-SPLIT|REDTEAM-AUDIT|CURRENT-MAIN-MECHANICS-V1` now. Any future Product-byte/spec change invalidates this SHA-bound verdict and requires fresh independent audit.
